relay-cardinality/src/redis/cardinality.lua

```diff
@@ -85,11 +85,16 @@ local results = {
 -- to all sets in `KEYS`.
 --
 -- Returns the total amount of values added to the 'working set'.
-local function sadd(t, offset, max)
+local function sadd(t, opts)
     local working_set_cardinality = 0
     local any_modifications = false
 
-    for i = 1, #KEYS do
+    opts = opts or {}
+    local key_offset = opts.skip_working_set and 1 or 0
+    local offset = opts.offset or 0
+    local max = opts.max
```

Comment:
do you also need default for `max` here ? Seems like it's not always provided.

Reply:
It's a bit of a mess, but `max` being `nil` is fine, it's checked in the batches function and has a different behaviour if it's not set.